src/integrationtest/java/com.microsoft.aad.msal4j/AuthorizationCodeIT.java

```diff
@@ -280,4 +313,80 @@ else if (authorityType == AuthorityType.ADFS){
 
         return app.getAuthorizationRequestUrl(parameters).toString();
     }
+
+    private String acquireAuthorizationCodeAutomatedWithClaims(
+            User user,
+            ClientApplicationBase app){
+
+        BlockingQueue<AuthorizationResult> authorizationCodeQueue = new LinkedBlockingQueue<>();
+
+        AuthorizationResponseHandler authorizationResponseHandler = new AuthorizationResponseHandler(
+                authorizationCodeQueue,
+                SystemBrowserOptions.builder().build());
+
+        httpListener = new HttpListener();
+        httpListener.startListener(8080, authorizationResponseHandler);
+
+        AuthorizationResult result = null;
+        try {
+            String url = buildAuthenticationCodeURLWithClaims(app);
+            seleniumDriver.navigate().to(url);
+            runSeleniumAutomatedLogin(user, app);
+
+            long expirationTime = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()) + 120;
+
+            while(result == null &&
+                    TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()) < expirationTime) {
+
+                result = authorizationCodeQueue.poll(100, TimeUnit.MILLISECONDS);
+            }
+        } catch(Exception e){
+            throw new MsalClientException(e);
+        } finally {
+            if(httpListener != null){
+                httpListener.stopListener();
+            }
+        }
+
+        if (result == null || StringHelper.isBlank(result.code())) {
+            throw new MsalClientException("No Authorization code was returned from the server",
+                    AuthenticationErrorCode.INVALID_AUTHORIZATION_RESULT);
+        }
+        return result.code();
+    }
+
+    private String buildAuthenticationCodeURLWithClaims(ClientApplicationBase app) {
+        String scope;
+        Set<String> claims = new HashSet<>();
+
+        //Two separate claims used to test merging claim JSONs
+        claims.add("{\"userinfo\":{\"given_name\":{\"essential\":true},\"nickname\":null,\"email\":{\"essential\":true},\"email_verified\":{\"essential\":true},\"picture\":null,\"http://example.info/claims/groups\":null}}");
```

Comment:
We briefly discussed this the other day, and after seeing it I think it would make sense to add these to the test constants file